# Session 7 - Conditional Probability & Bayes' Theorem

## Task 1 — P(Phone | Accessories)

Out of 500 users: 120 bought accessories, 80 bought a phone, 50 bought both.



In [1]:
total_users = 500
bought_accessories = 120
bought_phone = 80
bought_both = 50

p_accessories = bought_accessories / total_users
p_both = bought_both / total_users
p_phone_given_accessories = p_both / p_accessories

print('P(Accessories) =', p_accessories)
print('P(Phone and Accessories) =', p_both)
print('P(Phone | Accessories) =', p_phone_given_accessories)

P(Accessories) = 0.24
P(Phone and Accessories) = 0.1
P(Phone | Accessories) = 0.4166666666666667


## Task 2 — P(Smartphone | Headphones)

30% of users buy headphones, and 40% of those go on to buy a smartphone. That 40% figure is *already* the conditional probability being asked for — but the working below confirms it via the formula:

$$P(A\,|\,B) = \dfrac{P(A \text{ and } B)}{P(B)}$$

In [2]:
p_headphones = 0.3
p_smartphone_given_headphones = 0.4

p_joint = p_headphones * p_smartphone_given_headphones
p_check = p_joint / p_headphones

print('P(Headphones) =', p_headphones)
print('P(Headphones and Smartphone) = P(Headphones) x P(Smartphone|Headphones) =', p_joint)
print('P(Smartphone | Headphones) = P(joint) / P(Headphones) =', p_check)

P(Headphones) = 0.3
P(Headphones and Smartphone) = P(Headphones) x P(Smartphone|Headphones) = 0.12
P(Smartphone | Headphones) = P(joint) / P(Headphones) = 0.4


**Answer:** P(Smartphone | Headphones) = **0.4 (40%)** — consistent with the number given in the problem, confirmed via the formula rather than just restated.

## Task 3 — Bayes' Theorem: Churn Probability Given a 'We Miss You' Email

$$P(\text{Churn}\,|\,\text{Email}) = \dfrac{P(\text{Email}\,|\,\text{Churn}) \times P(\text{Churn})}{P(\text{Email})}$$

In [3]:
def bayes_churn_probability(prior_churn, prob_email_given_churn, prob_email):
    return (prob_email_given_churn * prior_churn) / prob_email

result = bayes_churn_probability(prior_churn=0.15, prob_email_given_churn=0.6, prob_email=0.25)
print('P(Churn | Email) =', result)

P(Churn | Email) = 0.36


**Interpretation:** A user's baseline churn risk is 15%, but among users who received the 'we miss you' email, 36% end up churning. The email is more common among users who were already drifting away, so receiving it is a meaningful signal — not proof the email *caused* churn.

## Task 4 — Bayes' Theorem: Fraud Detection

F = transaction is fraudulent, A = alert is triggered.
P(F) = 0.02, P(A|F) = 0.90 (true positive rate), P(A|not F) = 0.05 (false positive rate).

$$P(A) = P(A|F)P(F) + P(A|\text{not }F)P(\text{not }F)$$
$$P(F|A) = \dfrac{P(A|F)P(F)}{P(A)}$$

In [4]:
p_fraud = 0.02
p_alert_given_fraud = 0.90
p_alert_given_not_fraud = 0.05
p_not_fraud = 1 - p_fraud

p_alert = (p_alert_given_fraud * p_fraud) + (p_alert_given_not_fraud * p_not_fraud)
p_fraud_given_alert = (p_alert_given_fraud * p_fraud) / p_alert

print('P(Alert) =', p_alert)
print('P(Fraud | Alert) =', p_fraud_given_alert)

P(Alert) = 0.067
P(Fraud | Alert) = 0.26865671641791045


**Interpretation:** Even though the alert system catches 90% of real fraud, only about **26.9%** of flagged transactions actually turn out to be fraudulent. That's the classic base-rate effect: fraud is so rare (2% of all transactions) that the much larger pool of legitimate transactions still produces more false alarms (5% of 98% of traffic) than true positives (90% of a tiny 2% slice). A flag is a reason to check, not a verdict.

## Task 5 — AI-Generated Scenario: Conditional Probability in Food Delivery

**AI-generated scenario:** A food delivery app like Zomato or Swiggy can use conditional probability to power smarter recommendations. For example, the app can track that users who order biryani on a Friday night have a much higher probability of also adding a dessert to their cart than users placing a random weekday lunch order — i.e., P(dessert | Friday-night biryani order) is meaningfully higher than the app's overall P(dessert). Bayes' Theorem lets the recommendation engine go further: as new signals come in — say, a user clicks on a limited-time dessert offer, or the app runs a weekend dessert promotion — the system can update its prior belief about that user's dessert-purchase probability in real time (P(dessert | new signal, past orders)) instead of waiting to retrain a model from scratch, making the recommendation shown at checkout more personalized to that moment.

**My analysis:** This works because ordering behavior clearly isn't independent from context (day, time, past orders), so conditioning recommendations on that context should outperform showing every user the same generic 'people also bought' suggestions.